src/main/java/Duke/CommandList.java

```diff
@@ -0,0 +1,133 @@
+package Duke;
+
+import Task.Task;
+import Task.ToDo;
+import Task.Deadline;
+import Task.Event;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class CommandList {
+
+    /**
+     * Execute different actions based on user's input taken from Parser.
+     *
+     * @param input command to execute
+     * @param parser Parser object to terminate program if necessary
+     * @param list Arraylist of Tasks to manipulate
+     * @param storage Storage object to load/store list of Tasks
+     */
+    public void execute(String input, Parser parser, ArrayList<Task> list, Storage storage) {
+        String[] inputs = input.split(" ");
+        String firstWord = inputs[0];
+
+        if (input.equals("bye")) { // check for exit condition
+            System.out.println("Bye. Hope to see you again soon!");
+            parser.terminate();
+        } else if (input.equals("list")) { // list all tasks
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < list.size(); i++) {
+                System.out.println((i + 1) + "." + list.get(i));
+            }
+        } else if (firstWord.equals("mark") || firstWord.equals("unmark")) { // mark/unmark task
```

Comment:
It seems like it's possible for this part of the code to throw an exception (such as if secondword can't be converted to an int or if index goes outside the bounds of the ArrayList). Perhaps you can catch and handle these exceptions?